src/OmniSharp.DotNetTest/TestFrameworks/TestFramework.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace OmniSharp.DotNetTest.TestFrameworks
+{
+    public abstract class TestFramework
+    {
+        private static readonly ImmutableDictionary<string, TestFramework> s_frameworks;
+
+        static TestFramework()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, TestFramework>();
+
+            var nunit = new NUnitTestFramework();
+            var xunit = new XunitTestFramework();
+
+            builder.Add(nunit.Name, nunit);
+            builder.Add(xunit.Name, xunit);
+
+            s_frameworks = builder.ToImmutable();
+        }
+
+        public static TestFramework GetFramework(string name)
+        {
+            return s_frameworks.TryGetValue(name, out var result)
+                ? result
+                : null;
+        }
+
+        public static IEnumerable<TestFramework> GetFrameworks()
+        {
+            foreach (var kvp in s_frameworks)
+            {
+                yield return kvp.Value;
+            }
+        }
+
+        public abstract string FeatureName { get; }
+        public abstract string Name { get; }
+        public abstract string MethodArgument { get; }
+
+        protected abstract bool IsTestAttributeName(string typeName);
```

Comment:
Maybe, though I was expecting that we'd have to change this to support newer test frameworks.